cmd/release/release.go

```diff
@@ -0,0 +1,167 @@
+package release
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/buildsafedev/bsf/cmd/styles"
+	"github.com/buildsafedev/bsf/pkg/hcl2nix"
+	"github.com/buildsafedev/bsf/pkg/platformutils"
+	"github.com/buildsafedev/bsf/pkg/release"
+)
+
+var (
+	platform string
+)
+
+func init() {
+	ReleaseCmd.Flags().StringVarP(&platform, "platform", "p", "", "platform to release to")
+}
+
+// ReleaseCmd represents the release command
+var ReleaseCmd = &cobra.Command{
+	Use:   "release",
+	Short: "release the artifacts ",
+	Long: `
+	
+	GITHUB_ACCESS_TOKEN=xxx bsf release app v0.1
+	GITHUB_ACCESS_TOKEN=xxx bsf release app v0.1 --platform linux/amd64
+	`,
+	Args: cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println(styles.TextStyle.Render("Validating configuration..." + args[0]))
+
+		if len(args) < 2 {
+			fmt.Println(styles.ErrorStyle.Render("error: ", "missing arguments"))
+			fmt.Println(styles.HintStyle.Render("hint:", "run `bsf release app v0.1` to release the artifact"))
+			os.Exit(1)
+		}
+
+		accessToken := os.Getenv("GITHUB_ACCESS_TOKEN")
+		if accessToken == "" {
+			fmt.Println(styles.ErrorStyle.Render("error: ", "GITHUB_ACCESS_TOKEN is not set"))
+			fmt.Println(styles.HintStyle.Render("hint:", "run `export GITHUB_ACCESS_TOKEN=xxx` to set the access token"))
+			os.Exit(1)
+		}
+
+		data, err := os.ReadFile("bsf.hcl")
+		if err != nil {
+			fmt.Println(styles.ErrorStyle.Render("error: ", "failed to read config"))
+			fmt.Println(styles.HintStyle.Render("hint:", "run `bsf init` "))
+			os.Exit(1)
+		}
+
+		var dstErr bytes.Buffer
+		conf, err := hcl2nix.ReadConfig(data, &dstErr)
+		if err != nil {
+			fmt.Println(styles.ErrorStyle.Render("error: ", "failed to read config"))
+			fmt.Println(styles.HintStyle.Render("hint:", "run `bsf init` "))
+			os.Exit(1)
+		}
+
+		ghHCL, err := hcl2nix.ReadGitHubReleaseParams(conf, args[0])
+		if err != nil {
+			fmt.Println(styles.ErrorStyle.Render("error: ", err.Error()))
+			fmt.Println(styles.HintStyle.Render("hint:", "run `bsf init` "))
```

Comment:
this hint should not be here right?

Reply:
just thought someone might run this command before `bsf init` so we can give the hint. 
why do you think it doesn't belong here?

Reply:
the logic won't ever reach this point, right? it will fail here: https://github.com/buildsafedev/bsf/blob/58c40089834283627b126627a161cc5f6652893c/cmd/release/release.go#L53

Reply:
they could have done a `touch bsf.hcl` lol. 
Just being thorough with catching errors and returning hints. It might never happen but we never know...